core-processor/src/common.rs

```diff
@@ -195,6 +198,13 @@ pub enum JournalNote {
         /// Value amount
         value: u128,
     },
+    /// Store programs requested by user to be initialized later
+    StoreNewPrograms {
+        /// Code hash used to create new programs with ids in `candidates` field
+        code_hash: CodeHash,
+        /// Collection of program candidate ids and their init message ids.
+        candidates: Vec<(ProgramId, MessageId)>,
```

Comment:
In this case we can have numerous journal notes with same code hashes. I think it would be nice not to make additional allocations on `journal` vector. What do you think?

Reply:
Discussed, ok